File: Templates/TypeEditor/src/TemplateSolutions/VB.CustomTypeEditor/CustomControlLibrary.Protocol/GlobalUtilities/ThrowHelper.vb
Namespace Global.System.Runtime.CompilerServices

    Friend Module ThrowHelper
        ''' <summary>
        '''  Thows an <see cref="ArgumentNullException"/>.
        ''' </summary>
        ''' <remarks>
        '''  This is marked with NoInlining to ensure that the JIT can better inline calling code.
        ''' </remarks>
        ''' <exception cref="ArgumentNullException"></exception>
        <MethodImpl(MethodImplOptions.NoInlining)>
        Public Sub ThrowArgumentNullException(ByVal paramName As String)
            Throw New ArgumentNullException(paramName)
        End Sub

        ''' <summary>
        '''  Throws an <see cref="ArgumentException"/>.
        ''' </summary>
        ''' <remarks>
        '''  This is marked with NoInlining to ensure that the JIT can better inline calling code.
        ''' </remarks>
        ''' <exception cref="ArgumentException"></exception>
        <MethodImpl(MethodImplOptions.NoInlining)>
        Public Sub ThrowArgumentException(ByVal message As String, ByVal paramName As String)
            Throw New ArgumentException(message, paramName)
        End Sub

        ''' <summary>
        '''  Throws an <see cref="InvalidOperationException"/>.
        ''' </summary>
        ''' <remarks>
        '''  This is marked with NoInlining to ensure that the JIT can better inline calling code.
        ''' </remarks>
        ''' <exception cref="InvalidOperationException"></exception>
        <MethodImpl(MethodImplOptions.NoInlining)>
        Public Sub ThrowInvalidOperationException(ByVal message As String)
            Throw New InvalidOperationException(message)
        End Sub
    End Module

End Namespace

File: Samples/RootDesigner/Net/ClientServerSample/RootDesignerDemo.Protocol/GlobalUtilities/ThrowHelper.cs
using System;
using System.Diagnostics.CodeAnalysis;
using System.Runtime.CompilerServices;

namespace Microsoft.WinForms.Utilities.Shared;

internal static class ThrowHelper
{
    /// <summary>
    ///  Throws an <see cref="ArgumentNullException"/>.
    /// </summary>
    /// <remarks>
    ///  This is marked with NoInlining to ensure that the JIT can better inline calling code.
    /// </remarks>
    /// <exception cref="ArgumentNullException"></exception>
    [DoesNotReturn]
    [MethodImpl(MethodImplOptions.NoInlining)]
    public static void ThrowArgumentNullException(string? paramName)
        => throw new ArgumentNullException(paramName);

    /// <summary>
    ///  Throws an <see cref="ArgumentException"/>.
    /// </summary>
    /// <remarks>
    ///  This is marked with NoInlining to ensure that the JIT can better inline calling code.
    /// </remarks>
    /// <exception cref="ArgumentException"></exception>
    [DoesNotReturn]
    [MethodImpl(MethodImplOptions.NoInlining)]
    public static void ThrowArgumentException(string message, string? paramName)
        => throw new ArgumentException(message, paramName);

    /// <summary>
    ///  Throws an <see cref="InvalidOperationException"/>.
    /// </summary>
    /// <remarks>
    ///  This is marked with NoInlining to ensure that the JIT can better inline calling code.
    /// </remarks>
    /// <exception cref="InvalidOperationException"></exception>
    [DoesNotReturn]
    [MethodImpl(MethodImplOptions.NoInlining)]
    public static void ThrowInvalidOperationException(string message)
        => throw new InvalidOperationException(message);
}
